### ISSUE TITLE
Shared Element Transitions in Compose not working with Landscapist Coil

### ISSUE THREAD

**Issue by ErickSorto** [skydoves/landscapist#620]
**Please complete the following information:**
- Library Version [e.g. v1.0.0]
- Affected Device(s) [e.g. Samsung Galaxy s10 with Android 9.0]
 
**Describe the Bug:**

The image fades away and then reappears on the next screen instead of a smooth transition. 

**Expected Behavior:**

The image should smoothly transition from one screen to another.

**Comment by skydoves** [comment]
That looks weird. [Pokedex-Compose](https://github.com/skydoves/pokedex-compose) project also uses it with the shared element transition animation but it works as expected 🤔

**Comment by ErickSorto** [comment]
> That looks weird. [Pokedex-Compose](https://github.com/skydoves/pokedex-compose) project also uses it with the shared element transition animation but it works as expected 🤔

After testing the Pokedex app, the issue persists. The same issue is found in Pokedex-Compose, but it is subtle. 
The Pokemon disappears for half a second before the transition begins. It isn't a device issue because if you look at the GIFs in Pokedex-Compose repositories showing the transitions you can also see it disappear. 

Just like this example in my app:
https://github.com/user-attachments/assets/ed602a25-c3cd-4c48-ac69-79cca6b70f16

Pokedex transition bug example:

**Comment by skydoves** [comment]
Hey @ErickSorto, do you mean the problem is image flickering?

**Comment by ErickSorto** [comment]
> Hey @ErickSorto, do you mean the problem is image flickering?

Yeah the image flickers using CoilImage but when I use a regular AsynImage it does not flicker.

**Comment by skydoves** [comment]
I believe this issue is relevant from #632, but let me look into the details for better caching strategies.

**Comment by smithaaron** [comment]
I'm also experiencing what I believe is the same issue. This doesn't happen if I use Coil directly.

https://github.com/user-attachments/assets/7bd902c6-2a6d-4191-b130-c6ad3f56f153

**Comment by DmitryVoronko** [comment]
Have the same issue on current version - 2.6.1, it doesn't flicker when I use regular AsynImage from coil 3.3.0

**Comment by skydoves** [comment]
This issue should be fixed in the recent release, 2.9.5, especially `landscapist-image`.
